import.py

```diff
@@ -0,0 +1,281 @@
+  outputGeneratedCode(modelsFilters, "world/models.py", "Insert generated modelsFilters here")
+  outputGeneratedCode(modelsGeoFilters, "world/models.py", "Insert generated modelsGeoFilters here")
+  outputGeneratedCode(modelsSnuggetReturns, "world/models.py", "Insert generated modelsSnuggetReturns here")
+
+  outputGeneratedCode(adminModelImports, "world/admin.py", "Replace the next line with generated adminModelImports", replace=True)
+  outputGeneratedCode(adminLists, "world/admin.py", "Insert generated adminLists here", replace=True)
+  outputGeneratedCode(adminSiteRegistrations, "world/admin.py", "Insert generated adminSiteRegistrations here")
+
+  outputGeneratedCode(loadMappings, "world/load.py", "Insert generated loadMappings here")
+  outputGeneratedCode(loadPaths, "world/load.py", "Insert generated loadPaths here")
+  outputGeneratedCode(loadImports, "world/load.py", "Insert generated loadImports here")
+
+  outputGeneratedCode(viewsSnuggetMatches, "world/views.py", "Insert generated viewsSnuggetMatches here")
+
+  print("\n")
+
+
+
+def reprojectShapefile(f, inputDir, outputDir, srs):
+  original = os.path.join(inputDir, f)
+  reprojected = os.path.join(outputDir, f)
+  if os.path.exists(reprojected):
+    print("Skipping reprojection because this file has previously been reprojected.")
+  else:
+    print("Reprojecting to", srs)
+    ogrCmd = [
+      "ogr2ogr",
+      reprojected,
+      original,
+      "-t_srs", srs
+    ]
+    subprocess.call(ogrCmd)
+  return reprojected
+
+
+
+def simplifyShapefile(original, outputDir, tolerance):
+  simplified = os.path.join(outputDir, os.path.basename(original))
+  if os.path.exists(simplified):
+    print("Skipping simplification because this file has previously been reprojected.")
+  else:
+    print("Simplifying with tolerance", tolerance)
+    ogrCmd = [
+      "ogr2ogr",
+      simplified,
+      original,
+      "-simplify", tolerance
+    ]
+    subprocess.call(ogrCmd)
+    print(original, "simplified.")
+  return simplified
+
+
+
+def askUserForFieldNames(sf, stem):
+  fieldNames = [x[0] for x in sf.fields[1:]]
+  print("Found the following fields in the attribute table:")
+  print(str(fieldNames).strip("[").strip("]").replace("'",""))
+  print("Which would you like to use to look up snuggets by?")
+  keyField = False
+  while keyField not in fieldNames:
+    keyField = input(">> ")
+#TODO: make sure it really is OK for key & unique fields to be the same
+  print("Which is unique for each shape? (can be the same field as before)")
+  uniqueField = False
+  while uniqueField not in fieldNames:
+    uniqueField = input(">> ")
+  print("Generating code for", stem, "using the following fields:")
+  print("'" + keyField + "' to look up snuggets.")
+  print("'" + uniqueField + "' as the unique ID.")
+  return keyField, uniqueField
+
+
+
+def detectGeometryType(sf, stem):
+  shapeType = 0
+  i = 0
+  while shapeType == 0:
```

Comment:
you might want something like http://stackoverflow.com/questions/9868653/find-first-list-item-that-matches-criteria instead of this. So something like

next(shape for shape in sf.shapes() if shape.shapeType != 0)